jest.config.js

```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  moduleNameMapper: {
+    "^vue$": "vue/dist/vue.common.js"
+  },
+  moduleFileExtensions: ["js", "json", "vue"],
+  transform: {
+    ".*\\.(vue)$": "vue-jest",
+    "^.+\\.js$": "babel-jest"
+  },
+  collectCoverage: true,
+  collectCoverageFrom: [
+    "lib/**/*.js",
+    "public/js/**/*.{js,vue}",
+    "!public/js/dist/**",
+    "!**/node_modules/**"
+  ],
+  roots: ["lib", "public/js"],
+  setupFiles: ["./lib/testSetup.js"],
+  testURL: "http://localhost/",
+  husky: {
+    hooks: {
+      "pre-commit": "pretty-quick --staged"
+    }
+  },
+  globals: {
+    SIMULATING: process.env.SIMULATING === true ? true : false
```

Comment:
Maybe this will do it? And the same in `App.vue`
```suggestion
    SIMULATING: process.env.SIMULATING === "true" ? true : false
```

Reply:
Oh! That was actually a bug all along. The environment will always contain strings.

Reply:
[![unmask the villain](https://media.giphy.com/media/26DMUXQypGJaHhTsQ/giphy.gif)](https://media.giphy.com/media/26DMUXQypGJaHhTsQ/giphy.gif)

Reply:
So SIMULATING: process.env.SIMULATING === "true" ? true : false actually allows jest to evaluate if the app is running in SIMULATING mode or not.

@underscoredotspace @kleinfreund @mpaarating - my pushed commit (#4039a4d) shows this change a few others. You guys take a look when you get a minute and feel free to test that SIMULATING global variable in the App.spec.js test.

I would just change line 61 in App.spec.js to:

expect(vm.simulating).toEqual('test');

When you run SIMULATING=true npm t you will see the test will fail and show vm.simulating was expected to be true.

Then, run just npm t and you'll see see the test fail and show vm.simulating was expected to be false.

Thanks for all the help on this y'all! 😄

Reply:
This is looking good now.

- Running `npm test`: Both `vm.simulating` and `SIMULATING` evaluate to `false`
- Running `SIMULATING=true npm test`: Both `vm.simulating` and `SIMULATING` evaluate to `true`

You can add a `console.log` call to test this as well. No need to change the assertion to equal `test` to get the expected value. :)